docs/madr/decisions/046-meshservice-multizone.md

```diff
@@ -0,0 +1,359 @@
+Then it's up to a mesh operator if they want to sync everything by default or opt-in exported services.
+
+##### MeshTrafficPermission
+
+Similarly to what we do with reachable services (see [MADR 031](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/031-automatic-rechable-services.md)), 
+We could implicitly check if service can be consumed by proxy in other zone and only then sync the service.
+This however is difficult, we either need to require `kuma.io/zone` for it to work or check which proxies are in other zones which is expensive.
+
+### Replicated services use case
+
+As opposed to the current solution with `kuma.io/service`, redis in zone `east` is a different service than redis in zone `west`.
+There are two ways we could go about aggregating those services.
+
+#### MeshService with dataplaneTags selector applied on global CP
+
+To achieve replicated service, we define a MeshService on global CP.
+This MeshService is then synced to all zones.
+
+```yaml
+kind: MeshService
+metadata:
+  name: redis-xzh64d
+  namespace: kuma-system
+  labels:
+    team: db-operators
+    kuma.io/mesh: default
+    kuma.io/origin: global
+spec:
+  selector:
+    dataplaneTags: # it has no kuma.io/zone so it aggregates all redis from redis-system from all zones
+      app: redis
+      k8s.kuma.io/namespace: redis-system
+status:
+  ports:
+  - port: 1234
+    protocol: tcp
+  addresses:
+  - hostname: redis.redis-system.svc.cluster.global
+    status: Available
+    origin: global-generator
+  vips:
+  - ip: 241.0.0.1
+    type: Kuma
+  zones: # computed on global. The rest of status fields are computed on the zone
+  - name: east
+  - name: west
+```
+
+The disadvantages:
+* We need to do a computation of list of zones on global cp, because we don't sync all Dataplanes across all zones.
+  This can be expensive because global has list of all data plane proxies.
+* We wouldn't have a capability of apply "multizone" service on specific zone, because we don't have all DPPs in all zones.
+
+**Hostname and VIP management**
+We can provide hostnames by selecting by `kuma.io/origin` label.
+
+```yaml
+type: HostnameGenerator
+name: global-generator
+spec:
+  meshServiceSelector:
+    matchLabels:
+      kuma.io/origin: global
+  template: {{ label "kuma.io/display-name" }}.{{ label "k8s.kuma.io/namespace" }}.svc.cluster.global
+```
+
+#### MeshMultiZoneService
+
+If we assume that replicated service is aggregation of multiple services, we can define a specific object for it.
+`MeshMultiZoneService` selects services instead of proxies. 
+
+```yaml
+kind: MeshMultiZoneService
+metadata:
+  name: redis-xzh64d
+  namespace: kuma-system
+  labels:
+    team: db-operators
+    kuma.io/mesh: default
+spec:
+  meshServiceSelector: # we select services by labels, not dataplane objects
+    matchLabels:
+      k8s.kuma.io/service-name: redis
+      k8s.kuma.io/namespace: redis-system
```

Comment:
How can I treat a service that's replicated on Kubernetes and universal zones as the same? Do I just add these labels to the universal service? Is this an important use case?

Reply:
You could either add those labels to Universal service or have a separate common label(s) for both environments `my.org/service`. Of course, assuming that you trust this label, so not everyone can modify however they want